src/slot.rs

```diff
@@ -300,31 +322,41 @@ impl<Conn: Connection> Slot<Conn> {
         };
 
         #[cfg(feature = "probes")]
-        let id = usdt::UniqueId::new();
-        #[cfg(feature = "probes")]
-        probes::recycle__start!(|| (&id, backend.address.to_string()));
+        probes::recycle__start!(|| (
+            self.inner.pool_name.as_str(),
+            slot_id.as_u64(),
+            backend.address.to_string()
+        ));
 
         let result = tokio::time::timeout(timeout, connector.on_recycle(&mut conn)).await;
 
         let slot = self.inner.guarded.lock().unwrap();
         match result {
             Ok(Ok(())) => {
                 #[cfg(feature = "probes")]
-                probes::recycle__done!(|| &id);
+                probes::recycle__done!(|| (self.inner.pool_name.as_str(), slot_id.as_u64()));
                 event!(Level::TRACE, "Connection recycled successfully");
```

Comment:
It seems to me that it would probably be useful to make sure that the ID and name are also included in the `tracing` events, though that's not the primary goal of this PR.

Reply:
Sure thing, added in some instrumentation calls to avoid repeating ourselves too